src/main/java/testUI/AndroidUtils/ADBUtils.java

```diff
@@ -369,16 +376,57 @@ private static String getChromedriverVersion(String chromeVersion) {
             putLogWarn("Could not connect to \"https://chromedriver.storage.googleapis.com/\"");
             return "80";
         }
-        String chromeDriverVersion = "";  // ToDo it is not pulling the version. I just
-        String new_chrome_version = "";
-        for (String text : body.split(version)) {
-            if (text.contains("/chromedriver_mac64.zip")) {
-                new_chrome_version =
-                        version + text.split("/chromedriver_mac64\\.zip")[0];
+        String chromeDriverVersion = "";
+        String chromeName = "";
+        String Platform = System.getProperty("os.name").toLowerCase();
+        if (Platform.contains("mac")) {
+            System.out.println(System.getProperty("os.arch"));
+            if (System.getProperty("os.arch").toLowerCase().contains("aarch64")) {
+                chromeName = "mac-aarch64";
+            } else {
+                chromeName = "/chromedriver_mac64.zip";
+            }
+        } else if (Platform.contains("linux")) {
+            chromeName = "/chromedriver_linux64.zip";
+        } else {
+            chromeName = "/chromedriver_win32.zip";
+        }
+        Document doc = loadXMLFromString(body);
+        // Getting each of the chrome driver versions from the chromium api
+        NodeList list = doc.getDocumentElement().getElementsByTagName("Contents");
+
+        for (int temp = 0; temp < list.getLength(); temp++) {
+            Node node = list.item(temp);
+            if (node.getNodeType() == Node.ELEMENT_NODE) {
+                Element element = (Element) node;
+
+                String text = element.getTextContent();
```

Comment:
The current solution takes all text from all tags that are inside `Contents`, better use `<Key />` for faster string parsing
```suggestion
                String text = element.getElementsByTagName("Key").item(0).getTextContent();
```

Reply:
Im not sure if that works, but I will test it out